app/application.py

```diff
@@ -54,14 +56,21 @@ def drop_database():
 
 @app.before_request
 def before_request():
-    if request.endpoint is not None and 'health' not in request.endpoint and request.method != 'OPTIONS':
+    if _request_requires_authentication(request):
         log_request()
         res = authenticate(request.headers)
         if res != {'status': "ok"}:
             logger.debug("Failed to authenticate user")
             return res
 
 
+def _request_requires_authentication(req):
```

Comment:
No because request is a global. Any use of the name request will mask the global . So the temptation is to not pass it and rely on the global . However if you do that then the function becomes difficult and messy to test . Perhaps a different name